src/gyroaverage/gyroaverage_operator.hpp

```diff
@@ -0,0 +1,213 @@
+                ddc::DeviceAllocator<double>());
+        DFieldBSRTheta const coef = get_field(coef_alloc);
+        DConstFieldRTheta const rho_L = get_const_field(m_rho_L);
+
+        using SubDFieldRTheta = DField<
+                IdxRangeRTheta,
+                typename ExecutionSpace::memory_space,
+                Kokkos::layout_stride>;
+
+        ddc::for_each(batch_domain, [&](IdxBatch const ib) {
+            // FIXME
+            // The input of the spline builder must be LayoutRight
+            // We allocate a buffer in LayoutRight whereto the slice is copied
+            SubDFieldRTheta const sub_A = A[ib];
+            SubDFieldRTheta sub_A_bar = A_bar[ib];
+            DFieldMemRTheta sub_A_alloc(rtheta_mesh, ddc::DeviceAllocator<double>());
+            ddc::parallel_deepcopy(sub_A_alloc, sub_A);
+            spline_builder(coef, get_const_field(sub_A_alloc));
+
+            // Loop over r, theta
+            ddc::parallel_for_each(
+                    rtheta_mesh,
+                    KOKKOS_LAMBDA(IdxRTheta const irtheta) {
+                        IdxR const ir(irtheta);
+                        IdxTheta const itheta(irtheta);
+                        double const r = ddc::coordinate(ir);
+                        double const theta = ddc::coordinate(itheta);
+
+                        // Average over gyro points
+                        double tmp = 0.0;
+                        for (std::size_t igyro = 0; igyro < nb_gyro_points; igyro++) {
+                            // Compute the particle position in (R, Z) coordinate
+                            double const alpha = M_PI * 2.0 / static_cast<double>(nb_gyro_points)
+                                                 * static_cast<double>(igyro);
+                            double const R_p = r * Kokkos::cos(theta)
+                                               + rho_L(ir, itheta) * Kokkos::cos(alpha);
+                            double const Z_p = r * Kokkos::sin(theta)
+                                               + rho_L(ir, itheta) * Kokkos::sin(alpha);
+
+                            // Convert from (R, Z) into (r, theta) coordinate
+                            CoordRTheta p = coordinate_transform(R_p, Z_p);
+
+                            // Spline interpolation in (r, theta) coordinate
+                            tmp += spline_evaluator(p, get_const_field(coef));
+                        }
+                        sub_A_bar(ir, itheta) = tmp / static_cast<double>(nb_gyro_points);
+                    });
+
+            // Apply periodic boundary condition in theta direction
+            ddc::parallel_for_each(
+                    r_domain,
+                    KOKKOS_LAMBDA(IdxR const ir) {
+                        IdxTheta const theta_front = theta_domain.front();
+                        IdxTheta const theta_back = theta_domain.back();
+                        sub_A_bar(ir, theta_back) = sub_A_bar(ir, theta_front);
+                    });
```

Comment:
What do you think of
```suggestion
            ddc::deepcopy(sub_A_bar[theta_domain.back()], sub_A_bar[theta_domain.front()]);
```